cmd/init-repo/main.go

```diff
@@ -220,9 +246,10 @@ func cloneSourceRepo(cfg config.Config, runGodepRestore bool) {
 		return
 	}
 
-	repoLocation := fmt.Sprintf("https://%s/%s/%s", cfg.GithubHost, cfg.SourceOrg, cfg.SourceRepo)
+	repoLocation := fmt.Sprintf("https://%s/%s/%s.git", cfg.GithubHost, cfg.SourceOrg, cfg.SourceRepo)
```

Comment:
I don't think this was ever called because all the cloning process is done by the `hack/fetch-and-push-all.sh` script.

Also, this hard codes `BaseRepoPath` to `$GOPATH/src/k8s.io`, which should be fixed, but didn't want to change lots of things in this PR...